adagrams/game.py

```diff
@@ -1,11 +1,93 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+points = {
+      **dict.fromkeys(["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"], 1),
+      **dict.fromkeys(["D", "G"], 2),
+      **dict.fromkeys(["B", "C", "M", "P"], 3),
+      **dict.fromkeys(["F", "H", "V", "W", "Y"], 4),
+    "K" : 5,
+      **dict.fromkeys(["J", "X"], 8),
+      **dict.fromkeys(["Q", "Z"], 10)
+    }
+
 def draw_letters():
-    pass
+    letter_list = []
+    letter_count = {}
+    while len(letter_list) <10:
+        letter = random.choice(list(LETTER_POOL.keys())) 
+        if letter not in letter_count:
+            letter_count[letter] = 1
+        else:
+            letter_count[letter] +=1 
+        if letter_count[letter] <= LETTER_POOL[letter]:
+            letter_list.append(letter)
+    return letter_list
 
 def uses_available_letters(word, letter_bank):
-    pass
+    new_list = letter_bank.copy()
+    word_upper = word.upper()
+
+    for i in range(len(word)):
+        if word_upper[i] in new_list:
+            new_list.remove(word_upper[i])
+        else: 
+            return False 
+    return True 
 
 def score_word(word):
-    pass
+    score = 0
+    word_upper = word.upper()
+    if word == "":
+        return score
+    for letter in word_upper:
+        score += points[letter]
+    if len(word)>= 7:
+        score += 8
+    return score
+    
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    highest_word = ""
+    highest_score = 0
+    
+    for word in word_list:
+        score = score_word(word)
+        if score > highest_score:
+            highest_score = score
+            highest_word = word
+        elif score == highest_score:
+            if len(highest_word)!= 10:
```

Comment:
You could reduce the nesting here by combining these two `if` conditionals: 

```python
if len(highest_word) != 10 and (len(word) == 10 or len(word) < len(highest_word)):
```

Note that I inverted the second check because if `len(word) == 10`, Then there's no need to compare `len(word)` against `len(highest_word)` so we can take advantage of the short-circuiting behavior of the `or` operator.

But thinking about that, note that every call to `len` here represents a new call for something that isn't changing, and thus y'all are repeating that every time, which includes the work `len` takes to calculate the length.

Some helper variables here could save that work, and make it more readable:

```python
highest_length = len(highest_word)
if highest_length != 10:
    word_length = len(word)
    if word_length == 10 or word_length < highest_length:
        highest_word = word
```

Of course, now I've gone back to the very same nested `if` block structure y'all had before... But I want y'all to take that as yet another sign y'all were on the right path.